tests/system-tests/internal/apiobjectshelper/apiobjectshelper.go

```diff
@@ -81,3 +91,280 @@ func VerifyOperatorDeployment(apiClient *clients.Settings,
 
 	return nil
 }
+
+// DeleteDeployment deletes the deployment and verifies it and all related pods were removed.
+func DeleteDeployment(
+	apiClient *clients.Settings,
+	deploymentName, nsName, deploymentLabel string) error {
+	glog.V(100).Infof("Removing test deployment %q from %q ns", deploymentName, nsName)
+
+	if snifferDeployment, err := deployment.Pull(apiClient, deploymentName, nsName); err == nil {
+		glog.V(100).Infof("Deleting deployment %q from %q namespace", deploymentName, nsName)
+
+		err = snifferDeployment.DeleteAndWait(300 * time.Second)
+
+		if err != nil {
+			return fmt.Errorf("failed to delete deployment %q from %q namespace: %w", deploymentName, nsName, err)
+		}
+	} else {
+		glog.V(100).Infof("deployment %q not found in %q namespace", deploymentName, nsName)
+	}
```

Comment:
I have nothing to do with it since I do not have any idea how to differentiate such error.
I do not see any of implemented by you Delete... methods includes such verification and I never meet such failure before